crates/storage/src/connection/block.rs

```diff
@@ -115,24 +106,31 @@ impl Transaction<'_> {
 
         self.inner()
             .execute(
-                "DELETE FROM transactions WHERE block_number = ?",
+                "DELETE FROM block_headers WHERE number = ?",
                 params![&block],
             )
```

Comment:
Hmm, looks like we're not removing data from `transactions` in `purge_block()` anymore? Where does transaction data get pruned?

Reply:
Ah, no, those are fine. I wasn't aware that those `ON DELETE CASCADE`s are still present.